ext/pcre/php_pcre.c

```diff
@@ -545,15 +556,17 @@ static zend_string **make_subpats_table(uint32_t num_subpats, pcre_cache_entry *
 		return NULL;
 	}
 
-	subpat_names = ecalloc(num_subpats, sizeof(zend_string *));
+	subpat_names = pecalloc(num_subpats, sizeof(zend_string *), persistent);
 	while (ni++ < name_cnt) {
 		unsigned short name_idx = 0x100 * (unsigned char)name_table[0] + (unsigned char)name_table[1];
 		const char *name = name_table + 2;
-		subpat_names[name_idx] = zend_string_init(name, strlen(name), 0);
-		if (is_numeric_string(ZSTR_VAL(subpat_names[name_idx]), ZSTR_LEN(subpat_names[name_idx]), NULL, NULL, 0) > 0) {
```

Comment:
Just making sure: Was this code dead? PCRE seems to requrie:

> Compilation failed: subpattern name must start with a non-digit at offset 8

for digits, or:

> Compilation failed: subpattern name expected at offset 8

For symbols like `.`/`+`/`-` etc.

Reply:
This code is dead because it is checked during compilation of the regular expression and this check in php-src executes _after_ compilation. The compilation failure causes this code not to execute in the first place.